boards: others: add esp32c3_supermini board support

rlneumiller opened the pull request:
Having been unable to determine the original board
designer\vendor with any confidence, adding the
board under vendor "others".

Tested samples:
  * hello_world
  * basic/blinky
  
Signed-off-by: Arrel Neumiller <rlneumiller@gmail.com>

rlneumiller committed:
boards: others: add esp32c3_supermini board support

I have been unable to determine the original board
designer\vendor with any confidence so adding the
board under vendor "others".

Tested samples:
  * hello_world
  * basic/blinky

Signed-off-by: Arrel Neumiller <rlneumiller@gmail.com>

nordicjm commented on boards/others/esp32c3_supermini/doc/index.rst:
```diff
@@ -0,0 +1,282 @@
+Supported Features
+==================
+
+Current Zephyr's ESP32-C3-SUPERMINI board supports the following features:
```
put board name as used with west build here with \`\` around it and use board target instead of board

rlneumiller committed:
boards: others: Update index.rst supported features board name

Change board name to that used with west build as requested
by @nordicjm:
https://github.com/zephyrproject-rtos/zephyr/pull/78640#discussion_r1764899316

Signed-off-by: Arrel Neumiller <rlneumiller@gmail.com>

rlneumiller commented:
Made the mistake of using the github web UI to make edit as github profile name did not contain my full name, which I'm guessing is what github UI uses for "author". 
I've now updated github profile "name" to full name to match that in signed-off-by field.
Now waiting to see if ci will try again.

rlneumiller committed:
boards: others: esp32c3_supermini: minor update to trigger CI

This commit makes a minor whitespace change to trigger the CI pipeline.

rlneumiller commented:
Hmm... I thought that I had this process nailed but now I am experiencing other issues with the compliance checker in addition to my earlier issue involving use of the github UI to make an edit. 
I'll get the issues sorted and try again.
